python/ray/_private/runtime_env/packaging.py

```diff
@@ -409,19 +447,102 @@ def download_and_unpack_package(
                 with open(pkg_uri, "rb", transport_params=tp) as package_zip:
                     with open(pkg_file, "wb") as fin:
                         fin.write(package_zip.read())
+                unzip_package(pkg_file, local_dir, True, True, logger)
+            elif protocol == Protocol.GS or protocol == Protocol.HTTPS:
+                # Download package via GS or HTTPS.
+                try:
+                    from smart_open import open
+                except ImportError:
+                    raise ImportError(
+                        "You must `pip install smart_open` "
+                        f"to fetch {protocol.value.upper()} URIs.")
+                with open(pkg_uri, "rb") as package_zip:
```

Comment:
can we unify this with s3 path ? looks like the only difference between this part and s3 is boto3 and we call into smart_open with transport_params, but everything else is the same